### PR TITLE
Add basic link conditioner

### PR DESCRIPTION
**Pull request by fhaynes**
This adds a basic link conditioner. I propose we leave it inline for now, and evaluate conditional compilation or Traits for 0.2.0.

### CODE REVIEW

**Comment by LucioFranco** [comment]
@fhaynes please rebase against `master`

**Comment by fhaynes** [comment]
@LucioFranco Do you mean squash into one commit? I am rebased against master already.

**Comment by LucioFranco** [comment]
@fhaynes there are bors and travis updates on master that I want to make sure are in this branch, before merge.

**Comment by fhaynes** [comment]
Ah, found em. Rebased and pushed.

File: src/net/udp.rs [hunk 1]
```diff
@@ -49,16 +52,23 @@ impl UdpSocket {
...
         let mut bytes_sent = 0;
-
-        for payload in packet_data.parts() {
-            bytes_sent += self
-                .socket
-                .send_to(&payload, addr)
-                .map_err(|io| NetworkError::from(NetworkErrorKind::IOError { inner: io }))?;
+        if let Some(link_conditioner) = &self.link_conditioner {
+            if link_conditioner.should_send() {
+                for payload in packet_data.parts() {
```
Comment (LucioFranco) [inline src/net/udp.rs]: might be good to move this out to its own function because its repeated below.

Comment (fhaynes) [reply src/net/udp.rs]: #78

**Pull request merged**